front-end/src/screens/Settings/address.tsx

```diff
@@ -0,0 +1,177 @@
+import React, { useState, useContext, useEffect } from 'react';
+import { Icon } from 'semantic-ui-react';
+import { web3Accounts, web3FromSource, web3Enable } from '@polkadot/extension-dapp';
+import { InjectedAccountWithMeta } from '@polkadot/extension-inject/types';
+import { stringToHex } from '@polkadot/util';
+import Identicon from '@polkadot/react-identicon';
+
+import getExtensionUrl from '../../util/getExtensionUrl';
+import { NotificationContext } from '../../context/NotificationContext';
+import { useAddressesQuery, useAddressLinkConfirmMutation, useAddressLinkStartMutation, useAddressUnlinkMutation } from '../../generated/graphql';
+import Button from '../../ui-components/Button';
+import { NotificationStatus } from '../../types';
+
+const APP = 'polkassembly';
+const NETWORK = 'kasuma';
+
+const Address = (): JSX.Element => {
+	const [linked, setLinked] = useState<{[key: string]: boolean}>({});
+	const [accounts, setAccounts] = useState<InjectedAccountWithMeta[]>([]);
+	const [addressLinkStartMutation] = useAddressLinkStartMutation({ context: { uri : process.env.REACT_APP_AUTH_SERVER_GRAPHQL_URL } });
+	const [addressLinkConfirmMutation] = useAddressLinkConfirmMutation({ context: { uri : process.env.REACT_APP_AUTH_SERVER_GRAPHQL_URL } });
+	const [addressUnlinkMutation] = useAddressUnlinkMutation({ context: { uri : process.env.REACT_APP_AUTH_SERVER_GRAPHQL_URL } });
+	const { data } = useAddressesQuery({ context: { uri : process.env.REACT_APP_AUTH_SERVER_GRAPHQL_URL } });
+	const { queueNotification } = useContext(NotificationContext);
+
+	useEffect(() => {
+		async function connect() {
+			await web3Enable(APP);
+
+			const allAccounts = await web3Accounts();
+
+			setAccounts(allAccounts);
+		}
+
+		connect();
+	}, []);
+
+	useEffect(() => {
+		const addresses = (data && data.addresses) || [];
+		addresses.forEach((address) => {
```

Comment:
What I meant was to use the following:
```suggestion
		addresses.forEach(( {address} ) => {
```

so that you don't use `address.address` in the following lines.

Reply:
![Screenshot from 2020-01-22 16-35-37](https://user-images.githubusercontent.com/874046/72889252-40ec2400-3d35-11ea-971e-8a4f1f4304fe.png)
getting this error on {address}

Reply:
ok makes sense, not a great fan of the double address but I can live with :)